### DISCUSSION TITLE
syriで出てくるエラーについて

### DISCUSSION THREAD

**Discussion by Sugiura7** [nig-sc/nigsc_homepage2#341] [Q&A] [answered]
いつもお世話になっております
syriというツールをランしてみたところ下のようなエラーが出ました。numpyや他のツールのバージョンがおかしいのかと思い、個別に環境を立ち上げた上でパッケージのバージョンを要求通りにしたのですが、うまく動きません（Python >=3.8 and the following packages: Cython-0.29.23, numpy-1.21.2, scipy-1.6.2, pandas-1.2.4, python-igraph-0.9.1, psutil-5.8.0, pysam-0.16.0.1, and matplotlib-3.3.4）。

syriはg++をコンパイラとして要求しているのですが、それが原因で動いていない可能性はありますでしょうか？
遺伝研のモジュールにgccがあると思うのですが、g++をコンパイラとして使う方法はありますか？

コンパイラ等を理解しきれていないため分かりにくい質問になっていると思いますが、分かることがありましたら回答よろしくお願いします。

`(syri) [sugiura@at137 plotsr]$ syri -c R511_China.bam -r newR511.fasta -q china2022genome.fasta -F B --prefix R511_China &
[2] 72887
(syri) [sugiura@at137 plotsr]$ Traceback (most recent call last):
  File "/home/sugiura/miniconda3/envs/syri/bin/syri", line 6, in <module>
    main(sys.argv[1:])
  File "/home/sugiura/miniconda3/envs/syri/lib/python3.9/site-packages/syri/scripts/syri.py", line 326, in main
    syri(args)
  File "/home/sugiura/miniconda3/envs/syri/lib/python3.9/site-packages/syri/scripts/syri.py", line 130, in syri
    from syri.synsearchFunctions import readCoords
  File "syri/pyxFiles/synsearchFunctions.pyx", line 1, in init syri.synsearchFunctions
ValueError: numpy.ndarray size changed, may indicate binary incompatibility. Expected 96 from C header, got 88 from PyObject`

**Comment by sc-fs** [comment, marked as answer]
パッケージの依存関係に問題があるように思われます。
再度、仮想環境を作成して見てください。

こちらで以下を参考にインストールしてみましたが、
手順の仮想環境作成時に依存関連のエラーが発生しましたので、
設定するリポジトリ名を追加して作業しました。
https://github.com/schneebergerlab/syri/tree/master

作成した環境では、エラー（ValueError: numpy.ndarray size changed）は発生しませんでした。

こちらで仮想環境を作成した手順は、以下となります。
```
qlogin -l s_vmem=12g -l mem_req=12g
conda create -n syri_env -c bioconda -c conda-forge syri
conda activate syri_env
cd tool
git clone https://github.com/schneebergerlab/syri.git
cd syri
python setup.py install
syri -h
```

追加パッケージとツール
```
pip install python-igraph
conda install -c bioconda minimap2
```

実行
`syri -c out.bam -r refgenome -q qrygenome -k -F B`

**Reply by Sugiura7** [reply]
お忙しい中、わざわざありがとうございます。
一度順を追って試してみたいと思います。よろしくお願いします。
